src/iOS/Services/iOSPushNotificationService.cs

```diff
@@ -14,13 +16,31 @@ public Task<string> GetTokenAsync()
             return Task.FromResult(NSUserDefaults.StandardUserDefaults.StringForKey(TokenSetting));
         }
 
-        public Task RegisterAsync()
+        public bool IsRegisteredForPush => UIApplication.SharedApplication.IsRegisteredForRemoteNotifications;
+
+        public async Task RegisterAsync()
         {
-            var userNotificationTypes = UIUserNotificationType.Alert | UIUserNotificationType.Badge |
-                UIUserNotificationType.Sound;
-            var settings = UIUserNotificationSettings.GetSettingsForTypes(userNotificationTypes, null);
-            UIApplication.SharedApplication.RegisterUserNotificationSettings(settings);
-            return Task.FromResult(0);
+            var tcs = new TaskCompletionSource<bool>();
+
+            var authOptions = UNAuthorizationOptions.Alert | UNAuthorizationOptions.Badge | UNAuthorizationOptions.Sound;
+            UNUserNotificationCenter.Current.RequestAuthorization(authOptions, (granted, error) =>
+            {
+                if (error != null)
+                {
+                    Console.WriteLine($"Push Notifications {error}");
```

Comment:
Let's change it to `Debug` instead.

Reply:
yes all of them should be Debug when this PR is finished, they are all over the place hehe. Once we get the green light on using the notifications-test hub I'll keep testing this and check if this has been fixed, then I'll clean this PR and commit it